Fatal Error: Class "um_ext\um_polylang\core\Fields" not found

When activating this plugin I'm getting

Fatal error: Uncaught Error: Class "um_ext\um_polylang\core\Fields" not found in [path-to-local-install]/wp-content/plugins/um-polylang-main/includes/core/class-um-polylang.php:61

Stack trace:
```
#0 [path-to-local-install]/wp-content/plugins/um-polylang-main/includes/core/class-um-polylang.php(46): UM_Polylang->fields()
#1 [path-to-local-install]/wp-content/plugins/um-polylang-main/includes/core/class-um-polylang.php(35): UM_Polylang->__construct()
#2 [path-to-local-install]/wp-content/plugins/ultimate-member/includes/class-init.php(153): UM_Polylang::instance()
#3 [path-to-local-install]/wp-content/plugins/um-polylang-main/um-polylang.php(78): UM->set_class('Polylang', true)
#4 [path-to-local-install]/wp-includes/class-wp-hook.php(310): um_polylang_init('')
#5 [path-to-local-install]/wp-includes/class-wp-hook.php(334): WP_Hook->apply_filters(NULL, Array)
#6 [path-to-local-install]/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#7 [path-to-local-install]/wp-settings.php(495): do_action('plugins_loaded')
#8[path-to-local-install]/wp-config.php(110): require_once('/Users/...')
#9 [path-to-local-install]/wp-load.php(50): require_once('/Users/...')
#10 [path-to-local-install]/wp-blog-header.php(13): require_once('/Users/...')
#11 [path-to-local-install]/index.php(17): require('/Users/...')
#12 {main}
  thrown in [path-to-local-install]/wp-content/plugins/um-polylang-main/includes/core/class-um-polylang.php on line 61
```

How to reproduce:
- LocalWP with fresh WordPress install (PHP 8.1.9, WordPress 6.3.1, nginx, MySQL 8.0)
- Polylang 3.4.5 (latest) with two languages installed (English + Spain)
- Ultimate Member 2.6.11 (latest) installed (no changes to any setting, created only the login, registration, user, etc. pages with automated "Create pages" button that plugin provided)
- Installed latest version of this plugin -> fatal error comes on activation

--- a/README.md
+++ b/README.md
@@ -1,9 +1,9 @@
 # Ultimate Member - Polylang
+
 Integrates the **Ultimate Member** community plugin with the **Polylang** multilingual plugin.
 
-__Note:__ This is a free extension created for the community. The Ultimate Member team does not provide any support for this extension.
-
 ## Key features
+
 - Ability to translate email templates.
 - Ability to translate bio (description) field in profile.
 - Proper permalinks for the Account and User (profile) pages.
@@ -13,6 +13,7 @@ __Note:__ This is a free extension created for the community. The Ultimate Membe
 __Note:__ This plugin requires the [Ultimate Member](https://wordpress.org/plugins/ultimate-member/) and [Polylang](https://uk.wordpress.org/plugins/polylang/) plugins to be installed first.
 
 ### Clone from GitHub
+
 Open git bash, navigate to the **plugins** folder and execute this command:
 
 `git clone --branch=main git@github.com:umdevelopera/um-polylang.git um-polylang`
@@ -20,9 +21,11 @@ Open git bash, navigate to the **plugins** folder and execute this command:
 Once the plugin is cloned, enter your site admin dashboard and go to _wp-admin > Plugins > Installed Plugins_. Find the "Ultimate Member - Polylang" plugin and click the "Activate" link.
 
 ### Install from ZIP archive
+
 You can install this plugin from the [ZIP archive](https://drive.google.com/file/d/175PVG6tLK7z1wcrAawFfQdTVIC071Eup/view) as any other plugin. Follow [this instruction](https://wordpress.org/support/article/managing-plugins/#upload-via-wordpress-admin).
 
 ## How to use
+
 Go to *wp-admin > Ultimate Member > Settings > Email* to translate email templates. Click the "+" icon unter the flag to translate a template for the needed language. The plugin saves translated email templates to locale subfolders in the theme. See details [here](https://docs.ultimatemember.com/article/1335-email-templates).
 
 Go to *wp-admin > Pages* to translate pages Account, Login, Members, Password Reset, Register, User. Click the "+" icon unter the flag to translate a page for the needed language. See details [here](https://docs.ultimatemember.com/article/1449-how-to-translate-plugin#forms).
@@ -31,7 +34,7 @@ Go to *wp-admin > Settings > Permalinks* and click the "Save Changes" button to 
 
 __Note:__ The "Post name" permalink structure is recommended.
 
-### Screenshots:
+### Screenshots
 
 Image - Translate email templates.
 ![UM Settings, Email (polylang)](https://github.com/umdevelopera/um-polylang/assets/113178913/65d14995-257d-4311-a93a-8f944ea12ba9)
@@ -42,7 +45,12 @@ Image - Translate pages.
 Image - Permalink settings.
 ![WP Settings, Permalink (default)](https://github.com/umdevelopera/um-polylang/assets/113178913/69be91c9-12dd-490c-9145-b163c5beb26d)
 
-## Related links:
+## Support
+
+This is a free extension created for the community. The Ultimate Member team does not provide any support for this extension. Open new [issue](https://github.com/umdevelopera/um-polylang/issues) if you face a problem.
+
+## Related links
+
 Ultimate Member home page: https://ultimatemember.com/
 
 Ultimate Member documentation: https://docs.ultimatemember.com/

--- a/includes/core/class-mail.php
+++ b/includes/core/class-mail.php
@@ -40,6 +40,7 @@ class Mail {
 
 	/**
 	 * Adding locale suffix to the "Subject Line" field.
+	 *
 	 * Example: change 'welcome_email_sub' to 'welcome_email_sub_de_DE'
 	 *
 	 * @since 1.0.0
@@ -106,28 +107,26 @@ class Mail {
 	 *
 	 * @since 1.0.0
 	 *
-	 * @global object $polylang The Polylang instance.
-	 *
 	 * @param  array $columns The Email table headers.
 	 * @return array
 	 */
 	public function email_table_columns( $columns ) {
-		global $polylang;
-		$languages = pll_languages_list();
 
+		$languages = pll_languages_list();
 		if ( count( $languages ) > 0 ) {
 
-			$flags_column = '';
+			$flags = '';
 			foreach ( $languages as $language ) {
-				$language      = $polylang->model->get_language( $language );
-				$flags_column .= '<span class="um-flag" style="margin:2px">' . $language->flag . '</span>';
+				$language = PLL()->model->get_language( $language );
+				$flag     = is_object( $language ) ? $language->flag : $language;
+				$flags   .= '<span class="um-flag" style="margin:2px">' . $flag . '</span>';
 			}
 
 			$new_columns = array();
 			foreach ( $columns as $column_key => $column_content ) {
 				$new_columns[ $column_key ] = $column_content;
 				if ( 'email' === $column_key && ! isset( $new_columns['pll_translations'] ) ) {
-					$new_columns['pll_translations'] = $flags_column;
+					$new_columns['pll_translations'] = $flags;
 				}
 			}
 
@@ -139,7 +138,6 @@ class Mail {
 
 
 	/**
-	 *
 	 * Add cell for the column 'translations' in the Email table.
 	 *
 	 * @since 1.0.0
@@ -153,36 +151,35 @@ class Mail {
 		foreach ( $email_notifications as &$email_notification ) {
 			$email_notification['pll_translations'] = '';
 			foreach ( $languages as $language ) {
-				$email_notification['pll_translations'] .= $this->email_table_cell_pll_translations( $email_notification['key'], $language );
+				$email_notification['pll_translations'] .= $this->email_table_link( $email_notification['key'], $language );
 			}
 		}
 
 		return $email_notifications;
 	}
 
+
 	/**
-	 * Get content for the cell of the column 'translations' in the Email table.
+	 * Get a link to Add/Edit email template for a certain language.
 	 *
-	 * @since  2.1.6
-	 *
-	 * @global object $polylang The Polylang instance.
+	 * @since   1.0.0
+	 * @version 1.0.3 Use locale instead of the code in the template path.
 	 *
 	 * @param  string $template The email template slug.
 	 * @param  string $code     Slug or locale of the queried language.
 	 * @return string
 	 */
-	public function email_table_cell_pll_translations( $template, $code ) {
-		global $polylang;
+	public function email_table_link( $template, $code ) {
 
-		$language = $polylang->model->get_language( $code );
+		$language = PLL()->model->get_language( $code );
 		$default  = pll_default_language();
-		$lang     = $code === $default ? '' : trailingslashit( $code );
+		$locale   = $code === $default ? '' : trailingslashit( $language->get_prop( 'locale' ) );
 
 		// theme location.
-		$template_path = trailingslashit( get_stylesheet_directory() . '/ultimate-member/email' ) . $lang . $template . '.php';
+		$template_path = trailingslashit( get_stylesheet_directory() . '/ultimate-member/email/' . $locale ) . $template . '.php';
 
 		// plugin location for default language.
-		if ( empty( $lang ) && ! file_exists( $template_path ) ) {
+		if ( empty( $locale ) && ! file_exists( $template_path ) ) {
 			$template_path = UM()->mail()->get_template_file( 'plugin', $template );
 		}
 
@@ -193,12 +190,10 @@ class Mail {
 			)
 		);
 
-		$language_name = is_object( $language ) ? $language->name : $code;
-
 		if ( file_exists( $template_path ) ) {
 
 			// translators: %s - language name.
-			$hint      = sprintf( __( 'Edit the translation in %s', 'polylang' ), $language_name );
+			$hint      = sprintf( __( 'Edit the translation in %s', 'polylang' ), $language->get_prop( 'name' ) );
 			$icon_html = sprintf(
 				'<a href="%1$s" title="%2$s" class="pll_icon_edit"><span class="screen-reader-text">%3$s</span></a>',
 				esc_url( $link ),
@@ -208,7 +203,7 @@ class Mail {
 		} else {
 
 			// translators: %s - language name.
-			$hint      = sprintf( __( 'Add a translation in %s', 'polylang' ), $language_name );
+			$hint      = sprintf( __( 'Add a translation in %s', 'polylang' ), $language->get_prop( 'name' ) );
 			$icon_html = sprintf(
 				'<a href="%1$s" title="%2$s" class="pll_icon_add"><span class="screen-reader-text">%3$s</span></a>',
 				esc_url( $link ),

--- a/includes/core/class-permalinks.php
+++ b/includes/core/class-permalinks.php
@@ -63,12 +63,10 @@ class Permalinks {
 	 *
 	 * @since 1.0.0
 	 *
-	 * @global object $polylang The Polylang instance.
 	 * @param  array $rules Rewrite rules.
 	 * @return array
 	 */
 	public function add_rewrite_rules( $rules ) {
-		global $polylang;
 
 		$languages = pll_languages_list();
 		$newrules  = array();
@@ -79,7 +77,7 @@ class Permalinks {
 			$account         = get_post( $account_page_id );
 
 			foreach ( $languages as $language ) {
-				if ( pll_default_language() === $language && $polylang->options['hide_default'] ) {
+				if ( pll_default_language() === $language && PLL()->options['hide_default'] ) {
 					continue;
 				}
 				$lang_post_id  = pll_get_post( $account_page_id, $language );
@@ -88,7 +86,7 @@ class Permalinks {
 				if ( isset( $account->post_name ) && isset( $lang_post_obj->post_name ) ) {
 					$lang_page_slug = $lang_post_obj->post_name;
 
-					if ( 1 === $polylang->options['force_lang'] ) {
+					if ( 1 === PLL()->options['force_lang'] ) {
 						$newrules[ $language . '/' . $lang_page_slug . '/([^/]+)/?$' ] = 'index.php?page_id=' . $lang_post_id . '&um_tab=$matches[1]&lang=' . $language;
 					}
 
@@ -103,7 +101,7 @@ class Permalinks {
 			$user         = get_post( $user_page_id );
 
 			foreach ( $languages as $language ) {
-				if ( pll_default_language() === $language && $polylang->options['hide_default'] ) {
+				if ( pll_default_language() === $language && PLL()->options['hide_default'] ) {
 					continue;
 				}
 				$lang_post_id  = pll_get_post( $user_page_id, $language );
@@ -112,7 +110,7 @@ class Permalinks {
 				if ( isset( $user->post_name ) && isset( $lang_post_obj->post_name ) ) {
 					$lang_page_slug = $lang_post_obj->post_name;
 
-					if ( 1 === $polylang->options['force_lang'] ) {
+					if ( 1 === PLL()->options['force_lang'] ) {
 						$newrules[ $language . '/' . $lang_page_slug . '/([^/]+)/?$' ] = 'index.php?page_id=' . $lang_post_id . '&um_user=$matches[1]&lang=' . $language;
 					}
 

--- a/includes/core/class-um-polylang.php
+++ b/includes/core/class-um-polylang.php
@@ -58,6 +58,7 @@ class UM_Polylang {
 	 */
 	public function fields() {
 		if ( empty( UM()->classes['um_polylang_fields'] ) ) {
+			require_once um_polylang_path . 'includes/core/class-fields.php';
 			UM()->classes['um_polylang_fields'] = new um_ext\um_polylang\core\Fields();
 		}
 		return UM()->classes['um_polylang_fields'];
@@ -71,6 +72,7 @@ class UM_Polylang {
 	 */
 	public function form() {
 		if ( empty( UM()->classes['um_polylang_form'] ) ) {
+			require_once um_polylang_path . 'includes/core/class-form.php';
 			UM()->classes['um_polylang_form'] = new um_ext\um_polylang\core\Form();
 		}
 		return UM()->classes['um_polylang_form'];
@@ -84,6 +86,7 @@ class UM_Polylang {
 	 */
 	public function mail() {
 		if ( empty( UM()->classes['um_polylang_mail'] ) ) {
+			require_once um_polylang_path . 'includes/core/class-mail.php';
 			UM()->classes['um_polylang_mail'] = new um_ext\um_polylang\core\Mail();
 		}
 		return UM()->classes['um_polylang_mail'];
@@ -97,6 +100,7 @@ class UM_Polylang {
 	 */
 	public function permalinks() {
 		if ( empty( UM()->classes['um_polylang_permalinks'] ) ) {
+			require_once um_polylang_path . 'includes/core/class-permalinks.php';
 			UM()->classes['um_polylang_permalinks'] = new um_ext\um_polylang\core\Permalinks();
 		}
 		return UM()->classes['um_polylang_permalinks'];
@@ -108,12 +112,10 @@ class UM_Polylang {
 	 *
 	 * @since 1.0.0
 	 *
-	 * @global object $polylang The Polylang instance.
 	 * @param  string $field Optional, the language field to return (@see PLL_Language), defaults to `'slug'`.
 	 * @return string|int|bool|string[]|PLL_Language The requested field or object for the current language, `false` if the field isn't set.
 	 */
 	public function get_current( $field = 'slug' ) {
-		global $polylang;
 
 		$lang = pll_current_language();
 		if ( isset( $_GET['lang'] ) ) {
@@ -122,7 +124,7 @@ class UM_Polylang {
 		if ( empty( $lang ) || 'all' === $lang ) {
 			$lang = substr( get_locale(), 0, 2 );
 		}
-		$language = $polylang->model->get_language( $lang );
+		$language = PLL()->model->get_language( $lang );
 
 		return is_object( $language ) ? $language->get_prop( $field ) : $lang;
 	}
@@ -156,16 +158,13 @@ class UM_Polylang {
 	/**
 	 * Check if Polylang is active.
 	 *
-	 * @since 1.0.0
+	 * @since   1.0.0
+	 * @version 1.0.3 Check for the PLL function.
 	 *
 	 * @return boolean
 	 */
 	public function is_active() {
-		if ( defined( 'POLYLANG_VERSION' ) ) {
-			global $polylang;
-			return isset( $polylang ) && is_object( $polylang );
-		}
-		return false;
+		return defined( 'POLYLANG_VERSION' ) && function_exists( 'PLL' );
 	}
 
 

--- a/readme.txt
+++ b/readme.txt
@@ -1,14 +1,16 @@
 === Ultimate Member - Polylang ===
+
 Author: umdevelopera
 Author URI: https://github.com/umdevelopera
 Plugin URI: https://github.com/umdevelopera/um-polylang
 Tags: ultimate member, polylang, multilingual
-Requires at least: 6.0
-Tested up to: 6.3
-Stable tag: 1.0.2
+Requires at least: 5.5
+Tested up to: 6.4
+Requires UM core at least: 2.6.8
+Tested UM core up to: 2.7.0
+Stable tag: 1.0.3
 License: GNU Version 2 or Any Later Version
 License URI: http://www.gnu.org/licenses/gpl-3.0.txt
-Requires UM core at least: 2.6.7
 
 == Description ==
 
@@ -20,13 +22,15 @@ Integrates the "Ultimate Member" community plugin with the "Polylang" multilingu
 * Ability to translate bio (description) field in profile.
 * Proper permalinks for the Account and User (profile) pages.
 
-== Installation ==
-
-You can install this plugin from the ZIP file as any other plugin. Follow this instruction: https://wordpress.org/support/article/managing-plugins/#upload-via-wordpress-admin
-
 = Documentation & Support =
 
 This is a free extension created for the community. The Ultimate Member team does not provide any support for this extension. Open new issue in the GitHub repository if you face a problem: https://github.com/umdevelopera/um-polylang/issues
+
+== Installation ==
+
+Download ZIP file from GitHub or Google Drive. You can find download links here: https://github.com/umdevelopera/um-polylang
+
+You can install this plugin from the ZIP file as any other plugin. Follow this instruction: https://wordpress.org/support/article/managing-plugins/#upload-via-wordpress-admin
 
 == Changelog ==
 

--- a/um-polylang.php
+++ b/um-polylang.php
@@ -1,14 +1,17 @@
 <?php
 /**
-	Plugin Name: Ultimate Member - Polylang
-	Plugin URI:  https://github.com/umdevelopera/um-polylang
-	Description: Integrates Ultimate Member with Polylang.
-	Version:     1.0.3-alpha1
-	Author:      umdevelopera
-	Author URI:  https://github.com/umdevelopera
-	Text Domain: um-polylang
-	Domain Path: /languages
-	UM version:  2.6.7
+ * Plugin Name: Ultimate Member - Polylang
+ * Plugin URI:  https://github.com/umdevelopera/um-polylang
+ * Description: Integrates Ultimate Member with Polylang.
+ * Author:      umdevelopera
+ * Author URI:  https://github.com/umdevelopera
+ * Text Domain: um-polylang
+ * Domain Path: /languages
+ *
+ * Version: 1.0.3
+ * UM version: 2.7.0
+ * Requires at least: 5.5
+ * Requires PHP: 5.6
  */
 
 if ( ! defined( 'ABSPATH' ) ) {
@@ -25,18 +28,12 @@ define( 'um_polylang_plugin', plugin_basename( __FILE__ ) );
 define( 'um_polylang_extension', $plugin_data['Name'] );
 define( 'um_polylang_version', $plugin_data['Version'] );
 define( 'um_polylang_textdomain', 'um-polylang' );
-define( 'um_polylang_requires', '2.6.2' );
+define( 'um_polylang_requires', '2.7.0' );
+
 
 // Activation script.
 if ( ! function_exists( 'um_polylang_activation_hook' ) ) {
 	function um_polylang_activation_hook() {
-		$version = get_option( 'um_polylang_version' );
-		if ( ! $version ) {
-			update_option( 'um_polylang_last_version_upgrade', um_polylang_version );
-		}
-		if ( um_polylang_version !== $version ) {
-			update_option( 'um_polylang_version', um_polylang_version );
-		}
 
 		// update core pages.
 		if ( function_exists( 'UM' ) && function_exists( 'pll_languages_list' ) ) {
@@ -49,6 +46,7 @@ if ( ! function_exists( 'um_polylang_activation_hook' ) ) {
 	}
 }
 register_activation_hook( um_polylang_plugin, 'um_polylang_activation_hook' );
+
 
 // Check dependencies.
 if ( ! function_exists( 'um_polylang_check_dependencies' ) ) {
@@ -73,12 +71,7 @@ if ( ! function_exists( 'um_polylang_check_dependencies' ) ) {
 			);
 		} else {
 			require_once 'includes/core/class-um-polylang.php';
-			function um_polylang_init() {
-				if ( function_exists( 'UM' ) ) {
-					UM()->set_class( 'Polylang', true );
-				}
-			}
-			add_action( 'plugins_loaded', 'um_polylang_init', 4, 1 );
+			UM()->set_class( 'Polylang', true );
 		}
 	}
 }

